src/main/kotlin/com/jetbrains/micropython/nova/actions.kt

```diff
@@ -259,28 +262,55 @@ class OpenMpyFile : ReplAction("Open file", true) {
     }
 }
 
-open class UploadFile() : DumbAwareAction("Upload File(s) to MicroPython Device") {
+open class UploadFile() : DumbAwareAction("Upload Item(s) to MicroPython Device") {
     override fun getActionUpdateThread(): ActionUpdateThread = BGT
 
     override fun update(e: AnActionEvent) {
         val project = e.project
-        val file = e.getData(CommonDataKeys.VIRTUAL_FILE)
-        if (project != null
-            && file?.isInLocalFileSystem == true
-            && ModuleUtil.findModuleForFile(file, project)?.microPythonFacet != null
-        ) {
-            e.presentation.text =
-                if (file.isDirectory) "Upload Directory to MicroPython Device" else "Upload File to MicroPython Device"
+        val files = e.getData(CommonDataKeys.VIRTUAL_FILE_ARRAY)
+        if (project != null && files != null) {
+            var directoryCount = 0
+            var normalFileCount = 0
+
+            for (file in files.iterator()) {
+                if (file == null || !file.isInLocalFileSystem || ModuleUtil.findModuleForFile(
+                        file,
+                        project
+                    )?.microPythonFacet == null
+                ) {
+                    return
+                }
+
+                if (file.isDirectory) {
+                    directoryCount++
+                } else {
+                    normalFileCount++
+                }
+            }
+
+            if (normalFileCount >= 1 && directoryCount >= 1) {
+                e.presentation.text = "Upload Items to MicroPython Device"
```

Comment:
This code is hard to read, and there are six branches for five cases.
probably it's easier to write like. Also normalFileCound sounds a bit misleading
```kotlin
if(fileCount  == 0) {
  if(directoryCount == 1) {...} else {...}
} else if(directoryCount == 0) {
  if(fileCount == 1) {...} else {...}
} else {...}
```

Reply:
I agree with you on renaming it to `fileCount` but are you sure that the existing conditional is too hard to read? It seems more readable to me than nesting them.

The last else condition might be unnecessary, I wasn't sure if it is possible to have a case where both file count and directory count equal to zero.

Reply:
four ifs in IMO two layers of ifs with no boolean operators, four conditions in total is easier than  five ifs with six conditions